Automattic/jetpack
# CRM: Ensuring users assigned to contacts linked to transactions, invoices, quotes can view those objects

Fixes https://github.com/Automattic/zero-bs-crm/issues/3443

## Proposed changes:

- This PR makes sure that users who are assigned to contacts are able to see the associated quotes, transactions and invoices linked to those contacts - if 'Assign Ownership' is unchecked from the CRM General Settings page but 'Contact Assignment' is checked.
- Notes: With these assignment settings, a user won't see an invoice, quote, or transaction to which they are not assigned, with the exception of unassigned transactions that only they have created.


### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/3443

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

You can replicate the issue by:

- Creating a couple of test contacts
- Creating an invoice, transaction, and quote linked to those contacts
- Creating a CRM Contact Manager user
- Making sure 'Contact assignment' is ticked on the General Settings page, and Assign Ownership is unticked: `wp-admin/admin.php?page=zerobscrm-plugin-settings`
- Assigning a contact to that user
- Logging in as that user, and attempting to view the transaction, quote or invoice linked to the assigned contact
- You will see 'You do not have permission to edit this [Transaction, Quote or Invoice]'

To test the fix:

- Apply this PR. Then follow with the above contacts / users and also transaction, quote and invoice, try viewing those again (as the user assigned to the contact). 
- You should be able to view them.
- Try viewing a different transaction, quote or invoice whose assigned contact you are not linked to, and you should see a permissions error.
